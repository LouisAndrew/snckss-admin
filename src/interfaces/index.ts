// import { Category } from './category'
// import { Brand } from './brand'
// import { Order } from './order'
// import { Product } from './product'
// import { Admin } from './admin'

// export { Category, Brand, Order, Product, Admin }
export * from './category'
export * from './brand'
export * from './order'
export * from './product'
export * from './admin'
