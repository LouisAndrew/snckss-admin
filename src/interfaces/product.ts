/**
 * Product interface -> product object to illustrate the real product on the firestore.
 * vars -> would contain ref to firestore doc.
 */
export interface Product {
        name: string
        desc: string
        PID: string
        price: string
        imgs: string[]
        available: boolean
        arrivingAt: Date
        timesPurchased: number
        multipleVars: boolean
        vars: Product[]
}
