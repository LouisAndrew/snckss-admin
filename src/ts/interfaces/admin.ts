export interface Admin {
        name: string
        uid: string
}
