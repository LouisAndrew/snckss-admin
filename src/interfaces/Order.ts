import { Product } from './product'

/**
 * interface to illustrate how Order works on firestore.
 * Used to view orders from customer and therefore proccess it.
 */
export interface Order {
        UID: string
        products: Product[]
        orderID: string
        timeOrdered: Date
}
