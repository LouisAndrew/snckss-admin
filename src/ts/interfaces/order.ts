import { Product } from './product'

export enum PaymentMethod {
        BCA = 1,
        Mandiri = 2,
}

/**
 * interface to illustrate how Order works on firestore.
 * Used to view orders from customer and therefore proccess it.
 */
export interface Order {
        // automated fields based on order
        UID: string
        products: Product[]
        orderID: string
        paymentMethod: PaymentMethod

        // user info starts here
        name: string
        telephoneNum: string
        address: string
        timeOrdered: Date

        // should be confirmed by admin
        sent: boolean
        paid: boolean
        trackingNum: string
}
