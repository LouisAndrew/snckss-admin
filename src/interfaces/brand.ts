import { Product } from './product'
import { Category } from './category'

/**
 * Brand interface to illustrate how it works on firestore.
 * Used to categorize products based on its brand.
 */
export interface Brand {
        name: string
        products: Product[]
        category: Category
}
