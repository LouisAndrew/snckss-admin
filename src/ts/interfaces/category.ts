import { Brand } from './brand'

/**
 * interface to illustrate how Category works on firestore db.
 * used to organize products and brands based on its category e.g snack and chocolate.
 */
export interface Category {
        name: string
        brands: Brand[]
}
