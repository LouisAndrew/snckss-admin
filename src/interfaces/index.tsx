import { Category } from './category'
import { Brand } from './brand'
import { Order } from './order'
import { Product } from './product'
import { Admin } from './admin'

export type { Category, Brand, Order, Product, Admin }
