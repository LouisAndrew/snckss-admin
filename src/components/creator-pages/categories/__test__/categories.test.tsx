import React, { Suspense } from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { FirebaseAppProvider } from 'reactfire'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import config from '../../../../lib/firebase-config'
import Categories from '..'
import { Category } from '../../../../interfaces/category'
import { Brand } from '../../../../interfaces/brand'

describe('categories element.', () => {
        const mockAllCategories: Category[] = []
        const mockAllBrands: Brand[] = []
        afterEach(cleanup)
        const el = (
                <FirebaseAppProvider firebaseConfig={config}>
                        <Suspense fallback={<></>}>
                                <Categories
                                        allBrands={mockAllBrands}
                                        allCategories={mockAllCategories}
                                />
                        </Suspense>
                </FirebaseAppProvider>
        )

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        // TODO
        it('renders correctly', () => {})

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
