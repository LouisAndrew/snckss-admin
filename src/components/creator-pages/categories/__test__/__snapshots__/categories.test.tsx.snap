// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories element. matches snapshot 1`] = `
<div
  className="categories"
>
  <div
    className="back"
    onClick={[Function]}
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M20 11v2H8l5.5 5.5l-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5L8 11h12z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <h1>
    Categories
  </h1>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="category-name"
    >
      Category Name
    </label>
    <input
      className="form-control"
      id="category-name"
      onChange={[Function]}
      placeholder="Add category name here!"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="select-brands"
    >
      Select Brands
    </label>
    <select
      className="form-control"
      id="select-brands"
      name="select"
      onChange={[Function]}
      value=""
    >
      <option
        value="default"
      >
        Select multiple
      </option>
    </select>
  </div>
  <button
    aria-label={null}
    className="submit-btn btn btn-primary btn-block"
    onClick={[Function]}
    type="button"
  >
    Post
  </button>
</div>
`;
